### PR TITLE
Aaron/mkt 570 design feedback on the design system medium priority

### PR DESCRIPTION
**Pull request by gnoha**
[Jira](https://j2health.atlassian.net/browse/MKT-570)

NavMenu: 

Screenshot 2024-11-27 at 4 30 48 PM

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -18,6 +18,12 @@ const preview: Preview = {
         date: /Date$/i,
       },
     },
+    options: {
+      storySort: (a, b) =>
+        a.id === b.id
+          ? 0
+          : a.id.localeCompare(b.id, undefined, { numeric: true }),
```
Comment (avegancafe) [inline .storybook/preview.tsx:25]: This is dope

File: src/components/select/Select.tsx [hunk 2]
```diff
@@ -18,6 +19,26 @@ export const Select = (props: SelectProps) => {
...
+        )
+      }
+
+      if (!props.options || (props.options && props.options.length === 0)) {
+        return <div className="px-3 py-1">No options</div>
+      }
+
+      return menu
+    },
+    [props.loading]
```
[the remark this answers is not in the document]
Comment (gnoha) [reply src/components/select/Select.tsx:39]: Ugh fine

File: src/components/navMenu/NavMenu.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    onClick: () => {
+      alert('Settings')
+    },
+  },
+  {
+    type: 'divider',
+  },
+  {
+    key: 'logout',
+    label: <a>Logout</a>,
```
[the remark this answers is not in the document]
Comment (gnoha) [reply src/components/navMenu/NavMenu.stories.tsx:81]: Ignoring since this is in a storybook context.

File: src/components/navMenu/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { NavMenu } from './NavMenu'
```
Comment (avegancafe) [inline src/components/navMenu/index.ts:1]: Do we also need to export this in the top-level `index.tsx`?

**Review by avegancafe — CHANGES_REQUESTED** [review]
Nice! Looking good overall, only q is about that top level `index.tsx`, but if it's already done then we're all good!

File: .storybook/preview.tsx (hunk 1 above)
Comment (gnoha) [reply .storybook/preview.tsx:25]: This is slightly annoying bc within a story, `Docs` don't come first..

File: src/components/navMenu/index.ts (hunk 4 above)
Comment (gnoha) [reply src/components/navMenu/index.ts:1]: Oh right, thanks!

File: .storybook/preview.tsx (hunk 1 above)
Comment (avegancafe) [reply .storybook/preview.tsx:25]: Ehh I'll take it, I think we could also define the name of the `docs` so we could name it something like `_Docs` haha

File: src/components/select/Select.tsx [hunk 5]
```diff
@@ -18,6 +19,26 @@ export const Select = (props: SelectProps) => {
...
+      if (props.loading) {
+        return (
+          <div className="flex items-center justify-between px-3 py-1">
+            <span>Loading...</span>
+            <Spin indicator={<LoadingOutlined spin />} size="small" />
+          </div>
+        )
+      }
+
+      if (!props.options || (props.options && props.options.length === 0)) {
```
[the remark this answers is not in the document]
Comment (gnoha) [reply src/components/select/Select.tsx:33]: If it is loading, it should return early so it's not an issue.

[the remark this answers is not in the document]
Comment (avegancafe) [reply src/components/select/Select.tsx:33]: That seems... pretty obvious haha, I'm surprised it suggested this

File: src/components/navMenu/index.ts (hunk 4 above)
Comment (gnoha) [reply src/components/navMenu/index.ts:1]: Done: https://github.com/j2-health/design-system/pull/51/files#diff-b211563a379985cf25e1ffc4e40fd9bcbb669c8d4c61f4088b32fcd351c2722b

**Review by avegancafe — APPROVED** [review]
🎉

**Pull request merged**